### PR TITLE
Improved tracing option parsing

### PR DESCRIPTION
**Pull request by karknu**
TraceSelection contains 30+ bools for enabling/disabling different
tracers. This turns a bug in the order of those fields in
traceConfigParser from a run time error into a compile time error.
Based on an idea from @coot .

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/Config.hs [hunk 1]
```diff
@@ -21,78 +28,185 @@ data TraceOptions
   | TracingOn TraceSelection
   deriving (Eq, Show)
 
+type TraceAcceptPolicy = ("TraceAcceptPolicy" :: Symbol)
```
Comment (coot) [inline cardano-node/src/Cardano/Tracing/Config.hs:31]: We might not need brackets here.

A vim macro to get rid of it :D
```
qqdi(va("-pq
```

Comment (karknu) [reply cardano-node/src/Cardano/Tracing/Config.hs:31]: We don't need them for 8.10.2. I just added them so that it compiles with 8.6.3.

Comment (coot) [reply cardano-node/src/Cardano/Tracing/Config.hs:31]: Hmm, I didn't noticed that `GHC` parsers changed.

File: cardano-node/src/Cardano/Tracing/Config.hs [hunk 2]
```diff
@@ -21,78 +28,185 @@ data TraceOptions
...
+type TraceMempool = ("TraceMempool" :: Symbol)
+type TraceMux = ("TraceMux" :: Symbol)
+type TraceTxInbound = ("TraceTxInbound" :: Symbol)
+type TraceTxOutbound = ("TraceTxOutbound" :: Symbol)
+type TraceTxSubmissionProtocol = ("TraceTxSubmissionProtocol" :: Symbol)
+
+newtype OnOff (name :: Symbol) = OnOff { isOn :: Bool } deriving (Eq, Show)
+
+class TracerName tm where
+    getName :: tm -> Text
```
Comment (coot) [inline cardano-node/src/Cardano/Tracing/Config.hs:66]: @CodiePP you might now better if having such a class is useful for other context or not, if just for the `OnOff` type I would prefer just a function.

**Review by coot — APPROVED** [review]
LGTM

File: cardano-node/src/Cardano/Tracing/Config.hs [hunk 3]
```diff
@@ -21,78 +28,182 @@ data TraceOptions
...
+type TraceTxOutbound = ("TraceTxOutbound" :: Symbol)
+type TraceTxSubmissionProtocol = ("TraceTxSubmissionProtocol" :: Symbol)
+
+newtype OnOff (name :: Symbol) = OnOff { isOn :: Bool } deriving (Eq, Show)
+
+instance FromJSON (OnOff a) where
+    parseJSON (Data.Aeson.Bool b)= return $ OnOff b
+    parseJSON _ = mzero
+
+getName :: forall name. KnownSymbol name => OnOff name -> Text
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Tracing/Config.hs:69]: :100:

**Review by Jimbo4350 — APPROVED** [review]
:100:

**Review by intricate — APPROVED** [review]

**Review by CodiePP — APPROVED** [review]
I like it! :meetoooo:

**Pull request merged**